fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java

```diff
@@ -541,13 +542,16 @@ public void recalculateAccruals(Loan loan) {
 
     @Override
     public void setLoanDelinquencyTag(final Loan loan, final LocalDate transactionDate) {
-        // Revalidate the Delinquency Classification
-        final Long ageOfOverdueDays = loan.getAgeOfOverdueDays(transactionDate);
-        log.debug("Loan {} with {} days and current classification {}", loan.getId(), ageOfOverdueDays);
-        if (ageOfOverdueDays > 0L) { // If loan is overdue
-            this.delinquencyWritePlatformService.applyDelinquencyTagToLoan(loan, ageOfOverdueDays);
+        LoanScheduleDelinquencyData loanDelinquencyData = new LoanScheduleDelinquencyData(loan.getId(), loan.getLoanProduct().getId(),
+                transactionDate, null, loan);
+        loanDelinquencyData = this.delinquencyWritePlatformService.calculateDelinquencyData(loanDelinquencyData);
+        log.debug("Processing Loan {} with {} overdue days since date {}", loanDelinquencyData.getLoanId(),
+                loanDelinquencyData.getOverdueDays(), loanDelinquencyData.getOverdueSinceDate());
+        // Set or Unset the Delinquency Classification Tag
+        if (loanDelinquencyData.getOverdueDays() > 0) {
+            this.delinquencyWritePlatformService.applyDelinquencyTagToLoan(loanDelinquencyData.getLoan(), transactionDate);
         } else {
-            this.delinquencyWritePlatformService.removeDelinquencyTagToLoan(loan);
+            this.delinquencyWritePlatformService.removeDelinquencyTagToLoan(loanDelinquencyData.getLoan());
```

Comment:
Is the "loan" parameter and "loanDelinquencyData.getLoan()" is different?

Reply:
The Loan is passed in the loanDelinquencyData, so we don't need to load and reload the Loan in different parts